apis/v1alpha1/httproute_types.go

```diff
@@ -325,9 +325,13 @@ type HTTPRouteMatch struct {
 	// string for the group) or an implementation-defined resource (for
 	// example, resource "myroutematchers" in group "networking.acme.io").
 	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+	// group indicates that the resource is "configmaps".
+	//
+	// If the referent cannot be found, the route must be dropped
+	// from the Gateway. The controller should raise the "ResolvedRefs"
```

Comment:
This seems like it might be a bit too harsh. If someone has a large route and adds an invalid extension ref to one of many matches, we may not want to drop the entire route.

Reply:
Maybe. This introduces a new failure mode that we don't have any way to express - the route was accepted but some part of it was elided with unknown consequences.

Would we want to specify different error paths depending on the reference? For example, `.HTTPRouteSpec.ExtensionRef` seems like a good place to hang an external authorization configuration. If that fails to resolve, is it better to drop the route or program the route without authorization? Or should the controller make a decision based on the type and context of the reference?